packages/form-core/tests/FormApi.spec.ts

```diff
@@ -2907,4 +2908,147 @@ describe('form api', () => {
     expect(form.state.isValid).toBe(false)
     expect(form.state.canSubmit).toBe(true)
   })
+
+  it('should pass the current values to the standard schema when calling parseValuesWithSchema', async () => {
+    const nameSchema = z.object({
+      name: z.string(),
+    })
+    const firstNameSchema = z.object({
+      firstName: z.string().min(3),
+    })
+
+    const form = new FormApi({
+      defaultValues: {
+        firstName: '',
+      },
+    })
+    form.mount()
+
+    const nameSchemaResult = form.parseFieldValuesWithSchema(nameSchema)
+    // Name schema should complain that 'name' is missing in our form
+    expect(nameSchemaResult).not.toBeUndefined()
+    expect(Array.isArray(nameSchemaResult)).not.toBe(true)
+
+    expect(nameSchemaResult).toHaveProperty('fields')
+    expect(nameSchemaResult).toHaveProperty('form')
+
+    expect(nameSchemaResult?.fields).toHaveProperty('name')
+    expect(Array.isArray(nameSchemaResult?.fields['name'])).toBe(true)
+    expect(nameSchemaResult?.fields['name']).not.toHaveLength(0)
+
+    // First name schema should complain that 'firstName' is too short
+    const firstNameSchemaResult =
+      form.parseFieldValuesWithSchema(firstNameSchema)
+    expect(firstNameSchemaResult).not.toBeUndefined()
+    expect(Array.isArray(firstNameSchemaResult)).not.toBe(true)
```

Comment:
I'm curious, there are quite some `not` in all the new tests, do you see any paritcular advantage than just using `toBeDefined` or `toBe(false)` in some cases?